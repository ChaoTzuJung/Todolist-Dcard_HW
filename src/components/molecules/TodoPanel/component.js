import React from 'react';
import classnames from 'classnames';
import Button from 'components/atoms/Button';
import ButtonFile from 'components/atoms/ButtonFile';
import Field from 'components/atoms/Field';
import FieldDate from 'components/atoms/FieldDate';
import Icon from 'components/atoms/Icon';

import styles from './index.css';

const TodoPanel = ({
	className,
	imageUrl,
	hasImage,
	files,
	id,
	onUploadFile,
	handleChangeTextarea,
	textareaValue,
	...other
}) => (
	<div className={classnames(styles.todoPanel, className)}>
		<div className={styles.fieldGroup}>
			<label htmlFor={id}>
				<p>
					<Icon>date_range</Icon>
					<span>Deadline</span>
				</p>
				<div className={styles.row}>
					<FieldDate dateOnly id={id} {...other} />
					<FieldDate timeOnly id={id} {...other} />
				</div>
			</label>
			<label htmlFor={id}>
				<p>
					<Icon outlined>insert_drive_file</Icon>
					<span>File</span>
				</p>
				<div className={styles.row}>
					<ButtonFile onChange={onUploadFile}>
						<i className="material-icons" style={{ color: '#FFFFFF' }}>add</i>
					</ButtonFile>
					{hasImage && <span>{files.name}</span>}
				</div>
				{hasImage && (
					<div className={classnames(styles.photo, styles.preview)}>
						<img src={imageUrl} alt="preview" />
					</div>
				)}
			</label>

			<label htmlFor={id}>
				<p>
					<Icon outlined>textsms</Icon>
					<span>Comment</span>
				</p>
				<div className={styles.row}>
					<textarea
						type="text"
						id={id}
						name="comment"
						placeholder="Type your memo here..."
						onChange={handleChangeTextarea}
						value={textareaValue}
						{...other}
					/>
				</div>
			</label>
		</div>
		<div className={styles.buttonGroup}>
			<Button color="cancel">
				<Icon>close</Icon>
				Cancel
			</Button>
			<Button color="save">
				<Icon>add</Icon>
				Add Task
			</Button>
		</div>
	</div>
);

export default TodoPanel;
