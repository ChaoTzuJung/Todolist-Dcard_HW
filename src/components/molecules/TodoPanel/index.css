.todo-panel {
	position: relative;
	height: 100%;
	width: 620px;
	background: #F2F2F2;
	box-shadow: 0 4px 4px 0 #C8C8C8;
	border-top: 2px solid #C8C8C8;
	border-radius: 0 0 5px 5px;

	& > .field-group {
		padding: 24px 72px;

		& > .label {
			& > p {
				display: flex;
				align-items: center;

				& > span {
					margin-left: 6px;
					font-size: 20px;
    				font-weight: 500;
				}
			}

			& > .row {
				display: flex;
				align-items: center;

				& > :first-child {
					margin-left: 28px;
					margin-right: 8px;
				}

				&:nth-child(n) {
					&::placeholder {
						color: #CBCBCB;
						font-size: 16px;
						font-weight: 300;
					}
				}

				& > textarea {
					padding: 8px 16px;
					width: 445px;
					height: 120px;
					border: none;
					resize: none;

					&::placeholder {
						color: #CBCBCB;
						font-size: 16px;
						font-weight: 300;
						line-height: 24px;
					}
				}
			}
			& > .photo {
				margin-top: 24px;
				width: 100%;
				height: 100%;

				& > img {
					width: 100%;
				}
			}
		}
	}

	& > .button-group {
		display: flex;
	}
}
