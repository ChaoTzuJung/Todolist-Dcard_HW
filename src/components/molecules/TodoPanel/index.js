import { contain } from 'react-container-helper';

import TodoPanel from './component';

const initState = () => ({
	imageUrl: null,
	hasImage: false,
	files: null,
	textareaValue: '',
});

const mapSetStateToProps = (
	{ imageUrl, hasImage, files, textareaValue },
	{ className, id, onCancel, onSave },
	setState,
) => ({
	// state
	imageUrl,
	hasImage,
	files,
	textareaValue,

	// props
	className,
	id,
	onCancel,
	onSave,

	// action

	onUploadFile(e) {
		const file = e.target.files.item(0);
		console.log(file);
		const fr = new FileReader();
		fr.addEventListener('load', () => {
			setState({
				imageUrl: fr.result,
				hasImage: true,
			});
		});
		fr.readAsDataURL(file);
		setState({
			files: file,
		});
	},

	handleChangeTextarea(e) {
		setState({
			textareaValue: e.target.value,
		});
	},
	// submitFile() {
	// 	axios
	// 	axios.post('/img', imageUrl);
	// 	multpart
	// 	const form = new formData();
	// 	form.append(files)
	// 	axios.post('/img', {form});
	// },
});

export default contain(initState, mapSetStateToProps)(TodoPanel);
