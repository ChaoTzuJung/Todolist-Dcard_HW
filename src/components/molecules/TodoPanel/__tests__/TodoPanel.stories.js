import React from 'react';
import { withKnobs, boolean, text, select } from '@storybook/addon-knobs/react';

import TodoPanel from 'components/molecules/TodoPanel';

import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';

const stories = storiesOf('Commons|molecules/TodoPanel', module);

stories.addDecorator(withKnobs);

stories.add('__interactive', () => (
	<div style={{ margin: '100px' }}>
		<TodoPanel
			fileType={select(
				'file type',
				{ png: 'image/png', jpeg: 'image/jpeg', pdf: 'application/pdf', doc: 'application/zip' },
				'png',
			)}
			hasImage={boolean('hasImage', false)}
			fileName={text('fileName', '照片.png')}
			fileData={text(
				'imageUrl',
				'https://images.unsplash.com/photo-1517694712202-14dd9538aa97?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=800&q=60',
			)}
			textareaValue={text('comment', '內文')}
			onSave={action('Add Task')}
			onCancel={action('Cancel')}
		/>
	</div>
));
