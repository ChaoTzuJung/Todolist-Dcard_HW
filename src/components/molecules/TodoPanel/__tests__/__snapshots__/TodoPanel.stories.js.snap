// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`__interactive 1`] = `
<div
  style={
    Object {
      "margin": "100px",
    }
  }
>
  <TodoPanel
    onCancel={[Function]}
    onSave={[Function]}
    todoData={
      Object {
        "checked": false,
        "date": "",
        "edit": true,
        "fileData": "",
        "fileName": "",
        "fileType": "",
        "hasImage": false,
        "star": false,
        "startTime": "",
        "text": "",
        "textarea": "",
        "time": "",
      }
    }
  >
    <div
      className="todoPanel"
    >
      <div
        className="fieldGroup"
      >
        <div
          className="label"
        >
          <p>
            <Icon>
              <i
                className="icon material-icons"
              >
                date_range
              </i>
            </Icon>
            <span>
              Deadline
            </span>
          </p>
          <div
            className="row"
          >
            <Contain(FieldDate)
              dateOnly={true}
              handleDateChange={[Function]}
              todoData={
                Object {
                  "checked": false,
                  "date": "",
                  "edit": true,
                  "fileData": "",
                  "fileName": "",
                  "fileType": "",
                  "hasImage": false,
                  "star": false,
                  "startTime": "",
                  "text": "",
                  "textarea": "",
                  "time": "",
                }
              }
            >
              <FieldDate
                dateOnly={true}
                handleChange={[Function]}
                handleDateChange={[Function]}
                startDate={null}
                todoData={
                  Object {
                    "checked": false,
                    "date": "",
                    "edit": true,
                    "fileData": "",
                    "fileName": "",
                    "fileType": "",
                    "hasImage": false,
                    "star": false,
                    "startTime": "",
                    "text": "",
                    "textarea": "",
                    "time": "",
                  }
                }
              >
                <DatePicker
                  allowSameDay={false}
                  customInput={<InputComponent />}
                  dateFormat="yyyy/MM/dd"
                  dateFormatCalendar="LLLL yyyy"
                  disabled={false}
                  disabledKeyboardNavigation={false}
                  dropdownMode="scroll"
                  inlineFocusSelectedMonth={false}
                  minDate={2019-04-11T18:26:26.763Z}
                  monthsShown={1}
                  nextMonthButtonLabel="Next month"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onClickOutside={[Function]}
                  onFocus={[Function]}
                  onInputClick={[Function]}
                  onInputError={[Function]}
                  onKeyDown={[Function]}
                  onMonthChange={[Function]}
                  onSelect={[Function]}
                  onYearChange={[Function]}
                  placeholderText="yyyy/mm/dd"
                  preventOpenOnFocus={false}
                  previousMonthButtonLabel="Previous Month"
                  readOnly={false}
                  renderDayContents={[Function]}
                  selected={null}
                  shouldCloseOnSelect={true}
                  showMonthYearPicker={false}
                  showTimeInput={false}
                  showTimeSelect={false}
                  strictParsing={false}
                  timeCaption="Time"
                  timeInputLabel="Time"
                  timeIntervals={30}
                  withPortal={false}
                >
                  <PopperComponent
                    hidePopper={true}
                    popperComponent={null}
                    popperModifiers={
                      Object {
                        "preventOverflow": Object {
                          "boundariesElement": "viewport",
                          "enabled": true,
                          "escapeWithReference": true,
                        },
                      }
                    }
                    popperPlacement="bottom-start"
                    popperProps={Object {}}
                    targetComponent={
                      <div
                        className="react-datepicker__input-container"
                      >
                        <InputComponent
                          className=""
                          disabled={false}
                          onBlur={[Function]}
                          onChange={[Function]}
                          onClick={[Function]}
                          onFocus={[Function]}
                          onKeyDown={[Function]}
                          placeholder="yyyy/mm/dd"
                          readOnly={false}
                          value=""
                        />
                      </div>
                    }
                  >
                    <Manager>
                      <Reference>
                        <InnerReference
                          setReferenceNode={[Function]}
                        >
                          <div
                            className="react-datepicker-wrapper"
                          >
                            <div
                              className="react-datepicker__input-container"
                            >
                              <InputComponent
                                className=""
                                disabled={false}
                                onBlur={[Function]}
                                onChange={[Function]}
                                onClick={[Function]}
                                onFocus={[Function]}
                                onKeyDown={[Function]}
                                placeholder="yyyy/mm/dd"
                                readOnly={false}
                                value=""
                              >
                                <label
                                  className="label"
                                >
                                  <input
                                    className="input"
                                    disabled={false}
                                    onBlur={[Function]}
                                    onChange={[Function]}
                                    onClick={[Function]}
                                    onFocus={[Function]}
                                    onKeyDown={[Function]}
                                    placeholder="yyyy/mm/dd"
                                    readOnly={false}
                                    type="text"
                                    value=""
                                  />
                                </label>
                              </InputComponent>
                            </div>
                          </div>
                        </InnerReference>
                      </Reference>
                    </Manager>
                  </PopperComponent>
                </DatePicker>
              </FieldDate>
            </Contain(FieldDate)>
            <Contain(FieldDate)
              handleDateChange={[Function]}
              timeOnly={true}
              todoData={
                Object {
                  "checked": false,
                  "date": "",
                  "edit": true,
                  "fileData": "",
                  "fileName": "",
                  "fileType": "",
                  "hasImage": false,
                  "star": false,
                  "startTime": "",
                  "text": "",
                  "textarea": "",
                  "time": "",
                }
              }
            >
              <FieldDate
                handleChange={[Function]}
                handleDateChange={[Function]}
                startDate={null}
                timeOnly={true}
                todoData={
                  Object {
                    "checked": false,
                    "date": "",
                    "edit": true,
                    "fileData": "",
                    "fileName": "",
                    "fileType": "",
                    "hasImage": false,
                    "star": false,
                    "startTime": "",
                    "text": "",
                    "textarea": "",
                    "time": "",
                  }
                }
              >
                <DatePicker
                  allowSameDay={false}
                  customInput={<InputComponent />}
                  dateFormat="HH:mm"
                  dateFormatCalendar="LLLL yyyy"
                  disabled={false}
                  disabledKeyboardNavigation={false}
                  dropdownMode="scroll"
                  inlineFocusSelectedMonth={false}
                  minDate={2019-04-11T18:26:26.885Z}
                  monthsShown={1}
                  nextMonthButtonLabel="Next month"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onClickOutside={[Function]}
                  onFocus={[Function]}
                  onInputClick={[Function]}
                  onInputError={[Function]}
                  onKeyDown={[Function]}
                  onMonthChange={[Function]}
                  onSelect={[Function]}
                  onYearChange={[Function]}
                  placeholderText="HH:mm"
                  preventOpenOnFocus={false}
                  previousMonthButtonLabel="Previous Month"
                  readOnly={false}
                  renderDayContents={[Function]}
                  selected={null}
                  shouldCloseOnSelect={true}
                  showMonthYearPicker={false}
                  showTimeInput={false}
                  showTimeSelect={true}
                  showTimeSelectOnly={true}
                  strictParsing={false}
                  timeCaption="Time"
                  timeFormat="HH:mm"
                  timeInputLabel="Time"
                  timeIntervals={60}
                  withPortal={false}
                >
                  <PopperComponent
                    hidePopper={true}
                    popperComponent={null}
                    popperModifiers={
                      Object {
                        "preventOverflow": Object {
                          "boundariesElement": "viewport",
                          "enabled": true,
                          "escapeWithReference": true,
                        },
                      }
                    }
                    popperPlacement="bottom-start"
                    popperProps={Object {}}
                    targetComponent={
                      <div
                        className="react-datepicker__input-container"
                      >
                        <InputComponent
                          className=""
                          disabled={false}
                          onBlur={[Function]}
                          onChange={[Function]}
                          onClick={[Function]}
                          onFocus={[Function]}
                          onKeyDown={[Function]}
                          placeholder="HH:mm"
                          readOnly={false}
                          value=""
                        />
                      </div>
                    }
                  >
                    <Manager>
                      <Reference>
                        <InnerReference
                          setReferenceNode={[Function]}
                        >
                          <div
                            className="react-datepicker-wrapper"
                          >
                            <div
                              className="react-datepicker__input-container"
                            >
                              <InputComponent
                                className=""
                                disabled={false}
                                onBlur={[Function]}
                                onChange={[Function]}
                                onClick={[Function]}
                                onFocus={[Function]}
                                onKeyDown={[Function]}
                                placeholder="HH:mm"
                                readOnly={false}
                                value=""
                              >
                                <label
                                  className="label"
                                >
                                  <input
                                    className="input"
                                    disabled={false}
                                    onBlur={[Function]}
                                    onChange={[Function]}
                                    onClick={[Function]}
                                    onFocus={[Function]}
                                    onKeyDown={[Function]}
                                    placeholder="HH:mm"
                                    readOnly={false}
                                    type="text"
                                    value=""
                                  />
                                </label>
                              </InputComponent>
                            </div>
                          </div>
                        </InnerReference>
                      </Reference>
                    </Manager>
                  </PopperComponent>
                </DatePicker>
              </FieldDate>
            </Contain(FieldDate)>
          </div>
        </div>
        <div
          className="label"
        >
          <p>
            <Icon
              outlined={true}
            >
              <i
                className="icon material-icons-outlined"
              >
                insert_drive_file
              </i>
            </Icon>
            <span>
              File
            </span>
          </p>
          <div
            className="row"
          >
            <ButtonFile
              onChange={[Function]}
            >
              <button
                className="buttonFile"
                onClick={[Function]}
                type="button"
              >
                <div>
                  <i
                    className="material-icons"
                    style={
                      Object {
                        "color": "#FFFFFF",
                      }
                    }
                  >
                    add
                  </i>
                </div>
              </button>
              <input
                className="inputFile"
                onChange={[Function]}
                type="file"
              />
            </ButtonFile>
            <span />
          </div>
        </div>
        <div
          className="label"
        >
          <p>
            <Icon
              outlined={true}
            >
              <i
                className="icon material-icons-outlined"
              >
                textsms
              </i>
            </Icon>
            <span>
              Comment
            </span>
          </p>
          <div
            className="row"
          >
            <textarea
              name="comment"
              onChange={[Function]}
              placeholder="Type your memo here..."
              type="text"
            />
          </div>
        </div>
      </div>
      <div
        className="buttonGroup"
      >
        <Button
          color="cancel"
          onClick={[Function]}
        >
          <button
            className="button cancel"
            onClick={[Function]}
          >
            <div>
              <Icon>
                <i
                  className="icon material-icons"
                >
                  close
                </i>
              </Icon>
              Cancel
            </div>
          </button>
        </Button>
        <Button
          color="save"
          onClick={[Function]}
        >
          <button
            className="button save"
            onClick={[Function]}
          >
            <div>
              <Icon>
                <i
                  className="icon material-icons"
                >
                  add
                </i>
              </Icon>
              Add Task
            </div>
          </button>
        </Button>
      </div>
    </div>
  </TodoPanel>
</div>
`;
